src/NuGet.Core/NuGet.Packaging/Signing/ChainBuilding/X509ChainBuildPolicyFactory.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    internal static class X509ChainBuildPolicyFactory
+    {
+        // These fields are non-private only to facilitate testing.
+        internal const string EnvironmentVariableName = "NUGET_EXPERIMENTAL_CHAIN_BUILD_RETRY_POLICY";
+        internal const char ValueDelimiter = ',';
+
+        internal static IX509ChainBuildPolicy Create(IEnvironmentVariableReader reader)
+        {
+            if (reader is null)
+            {
+                throw new ArgumentNullException(nameof(reader));
+            }
+
+            if (RuntimeEnvironmentHelper.IsWindows)
+            {
+                string value = reader.GetEnvironmentVariable(EnvironmentVariableName);
+
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    return DefaultX509ChainBuildPolicy.Instance;
+                }
+
+                string[] parts = value.Split(ValueDelimiter);
+
+                if (parts.Length == 2
+                    && int.TryParse(parts[0], out int retryCount)
+                    && retryCount > 0
```

Comment:
should we enforce a `maximum retry count` so that we can handle edge cases such as `integer` max value? This comment applies to the timespan also.

Reply:
I feel like that's overengineering this workaround.  For the same reason I do not provide "reasonable" retry default values, I didn't enforce "reasonable" upper limits separate from the limits these types already enforce.  We could pick some values, raise error messages, and add some tests, but I don't think all that effort adds much value.